src/Sulu/Component/Rest/ListBuilder/Expression/Doctrine/AbstractDoctrineExpression.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Expression\Doctrine;
+
+use Doctrine\ORM\QueryBuilder;
+use Sulu\Component\Rest\ListBuilder\Expression\ExpressionInterface;
+
+/**
+ * Abstract definition for expressions used by the DoctrineListbuilder.
+ */
+abstract class AbstractDoctrineExpression implements ExpressionInterface
+{
+    /**
+     * Returns a statement for an expression.
+     *
+     * @param QueryBuilder $queryBuilder
+     *
+     * @return string
+     */
+    abstract public function getStatement(QueryBuilder $queryBuilder);
```

Comment:
Not so easy after all :-). During the lifetime of a DoctrineListbuilder three different instances of the doctrine QueryBuilder are created and the expression statements will be created two times with with two different instances. I think this PR is already big enough - I will add an issue and solve this when we integrate the filters for the SnippetBundle.